public/app/core/navigation/GrafanaRoute.tsx

```diff
@@ -14,9 +14,9 @@ export interface Props extends Omit<GrafanaRouteComponentProps, 'queryParams'> {
 export function GrafanaRoute(props: Props) {
   const { chrome } = useGrafana();
 
-  useEffect(() => {
-    chrome.routeMounted(props.route);
+  chrome.registerRouteRender(props.route);
```

Comment:
this needs to not be in an effect so that it happens before Page renders/mounts.